controllers/argocd/status.go

```diff
@@ -212,3 +217,44 @@ func (r *ReconcileArgoCD) reconcileStatusServer(cr *argoprojv1a1.ArgoCD) error {
 	}
 	return nil
 }
+
+// reconcileStatusHost will ensure that the host status is updated for the given ArgoCD.
+func (r *ReconcileArgoCD) reconcileStatusHost(cr *argoprojv1a1.ArgoCD) error {
+
+	if cr.Spec.Server.Route.Enabled {
+		if !IsRouteAPIAvailable() {
+			log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
+			return nil
+		}
+		route := newRouteWithSuffix("server", cr)
+		if argoutil.IsObjectFound(r.Client, cr.Namespace, route.Name, route) {
+			if route.Status.Ingress == nil {
+				status := "Pending"
+				cr.Status.Host = status
+				cr.Status.Phase = status
+				return r.Client.Status().Update(context.TODO(), cr)
+			} else {
+				cr.Status.Host = route.Spec.Host
+				cr.Status.Phase = "Available"
+				return r.Client.Status().Update(context.TODO(), cr)
+			}
+		}
+		return nil
+	} else if cr.Spec.Server.Ingress.Enabled {
+		ingress := newIngressWithSuffix("server", cr)
+		if argoutil.IsObjectFound(r.Client, cr.Namespace, ingress.Name, ingress) {
+			if ingress.Status.LoadBalancer.Ingress == nil {
+				status := "Pending"
+				cr.Status.Host = status
+				cr.Status.Phase = status
+				return r.Client.Status().Update(context.TODO(), cr)
+			} else {
+				cr.Status.Host = ingress.Spec.Rules[0].Host
```

Comment:
We should get hostnames from ingress.Status.LoadBalancer.LoadBalancerIngress which is a list.